rules/slack_rules/slack_user_privilege_escalation.py

```diff
@@ -13,16 +13,30 @@ def rule(event):
 
 
 def title(event):
-    if event.get("action") in USER_PRIV_ESC_ACTIONS:
-        return USER_PRIV_ESC_ACTIONS.get(event.get("action"))
+    username = deep_get(event, "actor", "user", "name", default="<unknown-actor>")
+    email = deep_get(event, "actor", "user", "email", default="<unknown-email>")
+
+    if event.get("action") == "owner_transferred":
+        return f"Slack Owner Transferred from {username} ({email})"
+
+    if event.get("action") == "permissions_assigned":
+        return f"Slack User, {username} ({email}), assigned permissions"
+
+    if event.get("action") == "role_change_to_admin":
+        return f"{username} ({email}) promoted to admin"
+
+    if event.get("action") == "role_change_to_owner":
+        return f"{username} ({email}) promoted to Owner"
+
     return "Slack User Privilege Escalation"
```

Comment:
```suggestion
    return "Slack User Privilege Escalation event {event.get('action')} on {username} ({email})"
```

Reply:
I like the human-friendly way of formatting the title that you've brought here. Perfectly fine to keep it as you have with a custom message for each. If you do that, Please add `Slack` as the first word to each title

Reply:
You got it! Change incoming.